types/index.d.ts

```diff
@@ -1102,6 +1102,10 @@ declare module 'cloudinary' {
 
             function restore(public_ids: string[], callback?: ResponseCallback): Promise<any>;
 
+             function restore_by_asset_ids(asset_ids: string[] | string, options?: AdminAndResourceOptions, callback?: ResponseCallback): Promise<ResourceApiResponse>;
```

Comment:
If I see correctly, the implementation of `restore_by_asset_ids` doesn't support passing a single value, we're only passing the first argument as a payload when doing the request to Admin API.
Currently Admin API only supports a list of values under `asset_ids` which means that, either this type definition needs to change to `string[]` or we should wrap the first argument in array if it's not already an array, in the `restore_by_asset_ids` implementation.